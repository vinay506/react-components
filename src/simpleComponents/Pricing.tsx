import React from 'react';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import PricePackCard from './PricePackCard/PricePackCard';
import styled from 'styled-components';
import reactIcon from '../images/react-icon.png';
import ContentOverCard from './ContentOverCard/ContentOverCard';
import ActivityCard from '../Dashboard/ActivityCard/ActivityCard';


function Pricing() {
    const clickMe = () => {
        console.log('clicked');
    }
    // <div style={{ width: '200px', height: '200px', border: '1px solid', position: 'relative' }}>
    //     <CustomBadge text={'Recommended'} position={'absolute'} top={'-10px'} bgColor={'grey'}></CustomBadge>
    // </div>

    // <h1>Pricing page</h1>
    //     <TextPrimary text='Packs to empower your preparation'></TextPrimary>
    //     <TextSecondary text='Packs & features'></TextSecondary>
    //     <div>
    //         <TextContent text='Compare between the features of packs, choose'></TextContent>
    //     </div>
    //     <TextBlur text="Current Pack"></TextBlur>
    //     <Link text='View Schedule' onLinkClick={clickMe} ></Link>
    //     <CustomButton type={BUTTON_TYPES.DENSE} text={'Get Started'} color={'blue'} bgColor={'white'} borderColor={'blue'} onButtonClick={clickMe}></CustomButton>
    const freePack = () => {
        console.log('free pack');
    }
    const secondPack = () => {
        console.log('second pack');
    }
    const recomendedPack = () => {
        console.log('recomonded pack');
    }

    const activityCardHandler = () => {
        console.log('Activity Card handler');
    }
    return (
        <PricingContainer>
            <br></br>
            <ActivityCard cardClick={activityCardHandler} logoImage={reactIcon} title="Tests" description="Take full, part or chapter tests to improve your performance.
"></ActivityCard>
            <br></br>

            <ContentOverCard>
                <div>
                    <TextPrimary text={'Packs to empower your preparation journey. '}></TextPrimary>
                    <TextPrimary text={'Cancel anytime with in 15 days.'}></TextPrimary>
                    <TextContent text={'Get Started with the pack best suited for you to prepare your upcoming events.'}></TextContent>
                </div>
            </ContentOverCard>
            <PricePackContainer>
                <PricePackCard titleText={'Free'}
                    packText={'Current pack'}
                    rupees={'0'} buttonText={'Get Started'}
                    buttonCallBack={freePack}></PricePackCard>
                <PricePackCard logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'}
                    buttonCallBack={secondPack}></PricePackCard>
                <PricePackCard badgeText={'Recommended'} logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'} buttonCallBack={recomendedPack}></PricePackCard>
            </PricePackContainer>
        </PricingContainer >
    )
}

const PricePackContainer = styled.div`
margin: 10px;
display: flex;
justify-content: center;
`
const PricingContainer = styled.div`
    text-align:center
`


export default Pricing
