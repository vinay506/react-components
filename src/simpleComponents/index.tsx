import TextSecondary from './TextSecondary/TextSecondary';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import TextBlur from './TextBlur/TextBlur';
import Link from './Link/Link';
import CustomBadge from './CustomBadge/CustomBadge';
import CustomButton from './CustomButton/CustomButton';
import CustomDropdown from './CustomDropdown/CustomDropdown';
import SquareCard from './SquareCard/SquareCard';
import RectangleCard from './RectangleCard/RectangleCard';
import RoundedBox from './RoundedBox/RoundedBox';
import CommentIcon from './Icons/CommentIcon/CommentIcon';
import NameWithSubText from './NameWithSubText/NameWithSubText';
import CustomTableComponent from './CustomTableComponent/CustomTableComponent';
import InfoIcon from './Icons/InfoIcon/InfoIcon';
import DisplayWithFlex from './DisplayWithFlex/DisplayWithFlex';
import Tooltip from './Tooltip/Tooltip';
import TextSecondaryRegular from './TextSecondaryRegular/TextSecondaryRegular';
import DateDetails from './DateDetails/DateDetails';

export {
    TextSecondary, TextPrimary, TextContent, TextBlur, Link, CustomBadge,
    CustomButton, CustomDropdown, SquareCard, RectangleCard, RoundedBox, CommentIcon,
    NameWithSubText,CustomTableComponent,InfoIcon,Tooltip,DisplayWithFlex,TextSecondaryRegular,
    DateDetails};
