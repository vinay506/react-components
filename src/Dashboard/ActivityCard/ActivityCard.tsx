import React from 'react'
import '@rmwc/card/styles';
import { Card } from '@rmwc/card';
import styled from 'styled-components'
import TextSecondaryRegular from '../../simpleComponents/TextSecondaryRegular/TextSecondaryRegular';
import { TextContent } from '../../simpleComponents';

const defaultCardStyles = {
    height: '104px',
    width: '333px',
    borderRadius: '8px',
    backgroundColor: '#FFFFFF',
    cursor: 'pointer',
    boxShadow: '0 8px 10px 1px rgba(0, 0, 0, 0.04), 0 3px 14px 2px rgba(0, 0, 0, 0.04), 0 5px 5px - 3px rgba(0, 0, 0, 0.04)'
};
function ActivityCard(props: any) {
    const { logoImage, title, description, cardClick } = props;
    return (
        <React.Fragment>
            <Card id="uniqueActivityCard" style={defaultCardStyles} onClick={cardClick}>
                <CardContentContainer>
                    <LeftContainer>
                        <LogoContiner>
                            <img src={logoImage} height="50" width="50" alt="Missing Logo" />
                        </LogoContiner>
                    </LeftContainer>
                    <RightContainer>
                        <TitleContainer>
                            <TextSecondaryRegular text={title}></TextSecondaryRegular>
                        </TitleContainer>
                        <DescriptionContainer>
                            <TextContent text={description}></TextContent>
                        </DescriptionContainer>
                    </RightContainer>
                </CardContentContainer>
            </Card>
        </React.Fragment>
    )
}

export default ActivityCard

const CardContentContainer = styled.div`
display: flex;
margin: 16px;
`
const LeftContainer = styled.div``

const RightContainer = styled.div`
  margin-left: 16px;
  `

const LogoContiner = styled.div`
padding: 11px 8px 11px 8px;
width: 64px;
border-radius: 8px;
background-color: #F6F5FA;
`

const TitleContainer = styled.div`
  height: 26px;
  width: 157px;
  color: #150F2D;
  font-family: "Amazon Ember";
  font-size: 17px;
  letter-spacing: 0;
  line-height: 24px;
  text-align: left;
  `

const DescriptionContainer = styled.div`
 height: 40px;
  width: 100%;
  color: #736F81;
  font-family: "Amazon Ember";
  font-size: 13px;
  letter-spacing: 0;
  line-height: 20px;
  text-align: left
  `
