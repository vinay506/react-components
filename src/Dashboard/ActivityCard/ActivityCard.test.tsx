import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import 'jest-styled-components'
import ActivityCard from './ActivityCard';
import reactIcon from '../../images/react-icon.png';
import { TextContent, TextSecondaryRegular } from '../../simpleComponents';

describe('Content Over Card', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultCardStyles: any;
    let imageStyles = {
        height: '50',
        width: '50'
    }
    const cardClickHandler = () => {
        return true;
    }
    beforeEach(() => {
        propInfo = {
            logoImage: reactIcon,
            title: 'Tests',
            description: 'Take full, part or chapter tests to improve your performance.',
            cardClick: cardClickHandler
        };
        defaultCardStyles = {
            height: '104px',
            width: '333px',
            borderRadius: '8px',
            backgroundColor: '#FFFFFF',
            cursor: 'pointer',
            boxShadow: '0 8px 10px 1px rgba(0, 0, 0, 0.04), 0 3px 14px 2px rgba(0, 0, 0, 0.04), 0 5px 5px - 3px rgba(0, 0, 0, 0.04)'
        };
        wrapper = shallow(<ActivityCard {...propInfo}></ActivityCard>);
    });


    // Snapshot validation
    test('Activity Card Snapshot', () => {
        const instance = renderer.create(<ActivityCard {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    // Props Validation
    test('Activity Card Element Validation', () => {
        const elementLength = wrapper.find('#uniqueActivityCard').length;
        expect(elementLength).toEqual(1);
    })

    test('Activity Click Check', () => {
        const button = wrapper.find('#uniqueActivityCard');
        const clickEvent = button.simulate('click');
        const returnValue = expect(clickEvent).toBeCalled;
        expect(returnValue).toBeTruthy();
    })

    test('Activity Card Title Element Check', () => {
        const elementLength = wrapper.find(TextSecondaryRegular).length;
        expect(elementLength).toEqual(1);
    })

    test('Activity Card Title Label Check', () => {
        const element = wrapper.find(TextSecondaryRegular);
        expect(element.props().text).toEqual(propInfo.title);
    })

    test('Activity Card Description Element Check', () => {
        const elementLength = wrapper.find(TextContent).length;
        expect(elementLength).toEqual(1);
    })

    test('Activity Card Description Label Check', () => {
        const element = wrapper.find(TextContent);
        expect(element.props().text).toEqual(propInfo.description);
    })

    test('Activity Card Image Element Validation', () => {
        const imgElement = wrapper.findWhere((obj: any) => obj.type() === 'img');
        expect(imgElement.length).toEqual(1);
    })

    test('Activity Card Image Source Validation', () => {
        const imgElement = wrapper.findWhere((obj: any) => obj.type() === 'img');
        expect(imgElement.props().src).toEqual(reactIcon);
    })

    test('Activity Card Image Height Validation', () => {
        const imgElement = wrapper.findWhere((obj: any) => obj.type() === 'img');
        expect(imgElement.props().height).toEqual(imageStyles.height);
    })

    test('Activity Card Image Width Validation', () => {
        const imgElement = wrapper.findWhere((obj: any) => obj.type() === 'img');
        expect(imgElement.props().width).toEqual(imageStyles.width);
    })
    // default styles validation
    test('Link Default Width Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.width).toEqual(defaultCardStyles.width);
    })

    test('Link Default Height Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.height).toEqual(defaultCardStyles.height);
    })

    test('Link Default Border Radius Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.borderRadius).toEqual(defaultCardStyles.borderRadius);
    })

    test('Link Default Background Color Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.backgroundColor).toEqual(defaultCardStyles.backgroundColor);
    })

    test('Link Default cursor Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.cursor).toEqual(defaultCardStyles.cursor);
    })

    test('Link Default Box Shadow Validation', () => {
        const element = wrapper.find('#uniqueActivityCard');
        expect(element.props().style.boxShadow).toEqual(defaultCardStyles.boxShadow);
    })

})