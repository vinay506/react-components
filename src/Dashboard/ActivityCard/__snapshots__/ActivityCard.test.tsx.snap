// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Over Card Activity Card Snapshot 1`] = `
.c4 {
  font-size: 18px;
  color: black;
  font-family: 'AmazonEmber-regular';
  -webkit-letterSpacing: 0;
  -moz-letterSpacing: 0;
  -ms-letterSpacing: 0;
  letterSpacing: 0;
  lineHeight: 24px;
}

.c6 {
  color: darkgrey;
  font-size: 13px;
  font-family: 'AmazonEmber-Regular';
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px;
}

.c2 {
  margin-left: 16px;
}

.c1 {
  padding: 11px 8px 11px 8px;
  width: 64px;
  border-radius: 8px;
  background-color: #F6F5FA;
}

.c3 {
  height: 26px;
  width: 157px;
  color: #150F2D;
  font-family: "Amazon Ember";
  font-size: 17px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 24px;
  text-align: left;
}

.c5 {
  height: 40px;
  width: 100%;
  color: #736F81;
  font-family: "Amazon Ember";
  font-size: 13px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 20px;
  text-align: left;
}

<div
  className="mdc-card"
  id="uniqueActivityCard"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": "8px",
      "boxShadow": "0 8px 10px 1px rgba(0, 0, 0, 0.04), 0 3px 14px 2px rgba(0, 0, 0, 0.04), 0 5px 5px - 3px rgba(0, 0, 0, 0.04)",
      "cursor": "pointer",
      "height": "104px",
      "width": "333px",
    }
  }
>
  <div
    className="c0"
  >
    <div
      className=""
    >
      <div
        className="c1"
      >
        <img
          alt="Missing Logo"
          height="50"
          src="react-icon.png"
          width="50"
        />
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
          id="textSecondaryRegular"
          style={
            Object {
              "color": undefined,
            }
          }
        >
          Tests
        </div>
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
          id="textContent"
          style={
            Object {
              "color": undefined,
            }
          }
        >
          Take full, part or chapter tests to improve your performance.
        </div>
      </div>
    </div>
  </div>
</div>
`;
